src/Reflex/Class.hs

```diff
@@ -225,6 +234,8 @@ class ( MonadHold t (PushM t)
   -- 'Behavior's and hold 'Event's, and use the results to produce a occurring
   -- (Just) or non-occurring (Nothing) result
   push :: (a -> PushM t (Maybe b)) -> Event t a -> Event t b
+  -- | Like 'push' but intended for functions that the implementation can consider cheap to compute for performance considerations.
```

Comment:
Let's put a bigger, scarier comment on this.  We need to make it extremely clear that the user's function may be run multiple times.